lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,39 +9,62 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache.
+	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
+	function wp_theme_has_theme_json() {
+		$cache_group       = 'theme_json';
+		$cache_key         = 'wp_theme_has_theme_json';
+		$theme_has_support = wp_cache_get( $cache_key, $cache_group );
```

Comment:
Hmm something looks off here. I know you had this in here before, but now it looks like this value is never returned, it's always overwritten in line 31, which means the cached value is ignored. Can you re-add the condition here to return this value early if it is set?

Reply:
Oh, I messed it up when removing the debug check. https://github.com/WordPress/gutenberg/pull/45543/commits/0aa8ce3d2ce997625f55b7673d96dac68213fab6 brings it back. Thanks for catching it.